--- a/src/client.h
+++ b/src/client.h
@@ -252,7 +252,8 @@
     WINDOW_TOOLBAR      = (1 << 5),
     WINDOW_MENU         = (1 << 6),
     WINDOW_UTILITY      = (1 << 7),
-    WINDOW_SPLASHSCREEN = (1 << 8)
+    WINDOW_SPLASHSCREEN = (1 << 8),
+    WINDOW_NOTIFICATION = (1 << 9)
 }
 netWindowType;
 
--- a/src/display.c
+++ b/src/display.c
@@ -155,6 +155,7 @@
         "_NET_WM_WINDOW_TYPE_SPLASH",
         "_NET_WM_WINDOW_TYPE_TOOLBAR",
         "_NET_WM_WINDOW_TYPE_UTILITY",
+        "_NET_WM_WINDOW_TYPE_NOTIFICATION",
         "_NET_WORKAREA",
         "MANAGER",
         "PIXMAP",
--- a/src/display.h
+++ b/src/display.h
@@ -249,6 +249,7 @@
     NET_WM_WINDOW_TYPE_SPLASH,
     NET_WM_WINDOW_TYPE_TOOLBAR,
     NET_WM_WINDOW_TYPE_UTILITY,
+    NET_WM_WINDOW_TYPE_NOTIFICATION,
     NET_WORKAREA,
     MANAGER,
     PIXMAP,
--- a/src/hints.h
+++ b/src/hints.h
@@ -82,6 +82,7 @@
 #define WIN_LAYER_DOCK                          8
 #define WIN_LAYER_ABOVE_DOCK                    10
 #define WIN_LAYER_FULLSCREEN                    12
+#define WIN_LAYER_NOTIFICATION                  14
 
 #define NET_WM_MOVERESIZE_SIZE_TOPLEFT          0
 #define NET_WM_MOVERESIZE_SIZE_TOP              1
--- a/src/netwm.c
+++ b/src/netwm.c
@@ -850,7 +850,8 @@
                 (atoms[i] == display_info->atoms[NET_WM_WINDOW_TYPE_DIALOG])  ||
                 (atoms[i] == display_info->atoms[NET_WM_WINDOW_TYPE_NORMAL])  ||
                 (atoms[i] == display_info->atoms[NET_WM_WINDOW_TYPE_UTILITY]) ||
-                (atoms[i] == display_info->atoms[NET_WM_WINDOW_TYPE_SPLASH]))
+                (atoms[i] == display_info->atoms[NET_WM_WINDOW_TYPE_SPLASH])  ||
+                (atoms[i] == display_info->atoms[NET_WM_WINDOW_TYPE_NOTIFICATION]))
             {
                 c->type_atom = atoms[i];
                 break;
@@ -1310,6 +1311,18 @@
                 XFWM_FLAG_HAS_MENU | XFWM_FLAG_HAS_MOVE |
                 XFWM_FLAG_HAS_RESIZE);
         }
+        else if (c->type_atom == display_info->atoms[NET_WM_WINDOW_TYPE_NOTIFICATION])
+        {
+            TRACE ("atom net_wm_window_type_notification detected");
+            c->type = WINDOW_NOTIFICATION;
+            c->initial_layer = WIN_LAYER_NOTIFICATION;
+            /* We unset these because CLIENT_FLAG_ABOVE will interfere with
+               our layer placement and put the window in the ABOVE_DOCK
+               layer, which is below the FULLSCREEN layer when the flags
+               are processed later. */
+            FLAG_UNSET (c->flags,
+                CLIENT_FLAG_ABOVE | CLIENT_FLAG_BELOW);
+        }
     }
     else
     {
